async/deadline.ts

```diff
@@ -8,7 +8,11 @@ import { abortable } from "./abortable.ts";
 
 /** Options for {@linkcode deadline}. */
 export interface DeadlineOptions {
-  /** Signal used to abort the deadline. */
+  /**
+   * Signal used to abort the deadline.
+   *
+   * @default {undefined}
```

Comment:
`undefined` will now show up in docs as the default value in quite a few of these cases. I consider that boilerplate and wanted to remove those instances before merging this. Does that all sound good, @kt3k?

Reply:
Sounds reasonable, but doesn't that cause a problem with the linter?

Reply:
Sounds good

> I'm not yet sure how else we can direct the linter to ignore a property.

Let's revisit that later